### ISSUE TITLE
[v2.0.0-alpha.3] Multiple mutations are not executed in a transaction (or are not rolled back)

### ISSUE THREAD

**Issue by nicolateral** [hasura/graphql-engine#6642]
Hasura does not handle multiple mutations in a transaction in v2.0.0-alpha.x.
If a mutation fail, all the executed mutations are not rolled back.

### Info
postgres:12
hasura/graphql-engine:v2.0.0-alpha.3

### Case
Here is a simple case to reproduce :

Two tables, user & comment are tracked by hasura.

```sql
CREATE TABLE public."user"
(
id uuid NOT NULL DEFAULT gen_random_uuid(),
name character varying NOT NULL,
CONSTRAINT user_pkey PRIMARY KEY (id)
);

CREATE TABLE public.comment
(
    id uuid NOT NULL DEFAULT gen_random_uuid(),
    user_id uuid NOT NULL,
    text text NOT NULL,
    CONSTRAINT comment_pkey PRIMARY KEY (id),
    CONSTRAINT comment_user_id_fkey FOREIGN KEY (user_id)
        REFERENCES public."user" (id) MATCH SIMPLE
        ON UPDATE RESTRICT
        ON DELETE RESTRICT
);

INSERT INTO public.user (id, name) VALUES ('7d7ca200-e938-4381-af4f-49a2942d3b63', 'Nicolas');
INSERT INTO public.comment(id, user_id, text) VALUES ('be846f9a-b489-4093-9533-3f7864f63f24', '7d7ca200-e938-4381-af4f-49a2942d3b63',  'My comment');
```

Send a mutation :

```graphql
mutation MyMutation {
  update_user(where: {id: {_eq: "7d7ca200-e938-4381-af4f-49a2942d3b63"}}, _set: {name: "Nicolas updated"}) {
    affected_rows
  }
  delete_user(where: {id: {_eq: "7d7ca200-e938-4381-af4f-49a2942d3b63"}}) {
    affected_rows
  }
}
```

Mutation response :
```json

{
  "errors": [
    {
      "extensions": {
        "path": "$",
        "code": "constraint-violation"
      },
      "message": "Foreign key violation. update or delete on table \"user\" violates foreign key constraint \"comment_user_id_fkey\" on table \"comment\""
    }
  ]
}
```

This seems fine but the first update user has not been rolled back (in the db, user "Nicolas" is now "Nicolas updated")

Send a query :
```graphql
query MyQuery {
  user(where: {id: {_eq: "7d7ca200-e938-4381-af4f-49a2942d3b63"}}) {
    id
    name
  }
}
```
Query response: 

```json
{
  "data": {
    "user": [
      {
        "id": "7d7ca200-e938-4381-af4f-49a2942d3b63",
        "name": "Nicolas updated"
      }
    ]
  }
}
```
I try to reproduce this case on hasura/graphql-engine:v1.3.3. v1.3.3 works well, as described in the documentation.

**Comment by abdullah2993** [comment]
This is a major hurdle for us and keeping us from using the `alpha` versions for development, Any ETA for this?

**Comment by paf31** [comment]
This was addressed in the 2.0.0-alpha.7 release and is documented at the top of the [release notes](https://github.com/hasura/graphql-engine/releases/tag/v2.0.0-alpha.7) there. For Postgres mutations specifically, the old behavior is now preserved.
